skietbaan_fe/src/components/Login.js

```diff
@@ -1,98 +1,135 @@
 import React, { Component } from 'react';
 import {
-  Container, Col, Form,
-  FormGroup, Label, Input,
-  Button,
+  Container, Col,
+  FormGroup, Label, Input,Button,Form
 } from 'reactstrap';
 import '../components/LoginStyles.css';
 
+  function validatePassword(str) {
+      var re = /(?=.*\d)(?=.*[a-z]).{6,}/;
+      return re.test(str);
+    }
+
+    function validateEmail(email) {
+      var re = /.+@.+\..+/;
+      return re.test(String(email).toLowerCase());
+    }
+
+    function validateUsername(username) {
+      var re = /[a-zA-Z]/;
+        return !re.test(String(username));
+    }
+    
 class App extends Component {
   constructor(props){
     super(props);
     this.state = {
       usernameValue : "",
       emailValue : "",
       passwordValue:"",
-      invalid:false
+      invalidPassword:false,
+      invalidEmail:false,
+      invalidUsername: false,
+      validForm:false
     }
-    this.showAlert = this.showAlert.bind(this);
+    this.Login = this.Login.bind(this);
     this.handleChange = this.handleChange.bind(this);
-    this.postData = this.handleChange.bind(this);
   }
   handleChange({ target }) {
     this.setState({
-      [target.name]: target.value
+      [target.name]: target.value,
+      invalidPassword:false,
+      invalidEmail: false,
+      invalidUsername: false
     });
-  }
-   postData(url = ``, data = {}) {
-    // Default options are marked with *
-      return fetch(url, {
-          method: "POST", // *GET, POST, PUT, DELETE, etc.
-          mode: "cors", // no-cors, cors, *same-origin
-          cache: "no-cache", // *default, no-cache, reload, force-cache, only-if-cached
-          credentials: "same-origin", // include, *same-origin, omit
-          headers: {
-              "Content-Type": "application/json",
-              // "Content-Type": "application/x-www-form-urlencoded",
-          },
-          redirect: "follow", // manual, *follow, error
-          referrer: "no-referrer", // no-referrer, *client
-          body: JSON.stringify(data), // body data type must match "Content-Type" header
-      })
-      .then(response => response.json()); // parses response to JSON
-  }
-  
-  showAlert() {
-
 
-    if(this.state.passwordValue === "" || this.state.passwordValue === null)
+    var Valid = true;
+    if(!validatePassword(this.state.passwordValue))
+    {
+        this.setState({
```

Comment:
Rather create a variable (e.g. `let stateUpdate = {}`) then at the end call setState passing this object. 
The reason behind this is because every time you call `this.setState()` it triggers a DOM rerender, having too many of these will lead to your app being slower. So it's good practice to try and minimize the number of times you have to rerender the DOM.

Reply:
I'm sorry i dont understand